src/coreclr/System.Private.CoreLib/src/System/Runtime/CompilerServices/RuntimeHelpers.CoreCLR.cs

```diff
@@ -877,8 +877,40 @@ public TypeHandle GetArrayElementTypeHandle()
         /// </summary>
         [MethodImpl(MethodImplOptions.InternalCall)]
         public extern MethodTable* GetMethodTableMatchingParentClass(MethodTable* parent);
+
+        /// <summary>
+        /// Given a statics pointer in the DynamicStaticsInfo, get the actual statics pointer.
+        /// </summary>
+        [MethodImpl(MethodImplOptions.InternalCall)]
+        public static extern ref byte MaskStaticsPointer(ref byte staticsPtr);
```

Comment:
```suggestion
        public static ref byte MaskStaticsPointer(ref byte staticsPtr)
        {
            fixed (byte* p = staticsPtr)
            {
                 return ref *(byte*)((nuint)p & ~(nuint)ISCLASSNOTINITED;
            }
        }
```
Can this be implemented like this in C# to avoid FCall? It is not on any perf critical path, so a few extra instructions from pinning are not a big deal.

Reply:
Good point. I had it on the critical path initially, and planned to make it an intrinsic to make it just work with it written as an fcall as a correct stopgap... then I realized I didn't need it on the critical path, and didn't move on from the FCALL.